Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,91 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        private readonly Stream source;
+        private StreamWriter? streamWriter;
+        private StreamReader? streamReader;
+
         public DataLoader(Stream source)
         {
+            this.source = source;
         }
 
-        public List<Mailbox> Load()
+        public List<Mailbox>? Load()
         {
+            source.Position = 0;
+            streamReader = new StreamReader(source);
```

Comment:
Since this is only used here, it could easily be declared as a local variable. 
```suggestion
            using var streamReader = new StreamReader(source, leaveOpen:true);
```